threatingestor/sources/github.py

```diff
@@ -8,29 +8,56 @@
 SEARCH_URL = "https://api.github.com/search/repositories"
 
 class Plugin(Source):
-
-    def __init__(self, name, search):
+    """Github Source Plugin"""
+    def __init__(self, name, search, username="", token=""):
         self.name = name
         self.search = search
 
+        if username and token:
+            self.auth = (username, token)
+        else:
+            self.auth = None
+
+    def make_requests(self, url, params, auth):
+        """Returns a list of artifacts"""
+        # Iterates through pages of results from query
+        artifact_list = []
+
+        response = requests.get(url, params=params, auth=self.auth)
```

Comment:
Let's use the `SEARCH_URL` constant here, and remove the `url` parameter from this method entirely. We can also remove the `auth` parameter from this method, since we access it as `self.auth` anyway.